CDCgov/prime-reportstream
# 5072 Updates to CO specific FHIR to HL7 schema

This PR has updates to the CO schema to satisfy the CO requirements provided to HCA.  Note that there is a requirement to fix bad quantity data that I have included to fix all bad numeric data. 

Test Steps:
1. Generate HL7 data using the CO schema from the HL7 data in `prime-router/src/testIntegration/resources/datatests/HL7_to_FHIR_to_HL7` for all the samples that the files DOES NOT have `expected` in the name.  Use the command `./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR_to_HL7/sample<name>.hl7 -s metadata/hl7_mapping/STLTs/CO/CO.yml --output-format HL7 --output-file build/sample<name>-expected-co.hl7`
2. Compare the original to the generated and verify the data looks good ignoring all the exceptions we know so far and verifying the data is manipulated per the CO requirements in https://cdc.sharepoint.com/:w:/r/teams/USDSatCDC/_layouts/15/Doc.aspx?sourcedoc=%7BB3D71AA4-7E47-4936-8403-BEAEB0A739F1%7D&file=MU_PH_CO_ELR_Requirements_ReportStream.docx&action=default&mobileredirect=true
3. Verify the build is successful in Github as it runs the new integration tests.

## Changes
- Added integration test to compare the raw files in the HL7 to FHIR to HL7 conversion
- Fixed a comparator issue in Quantity
- Allow Quantity to have non-valid string values, otherwise the HL7 to FHIR conversion blows up
- Added samples to integration test.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
